### PR TITLE
Add readbody logic in request completed state

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
This change is related to grpc intermittent failure in https://github.com/ballerina-platform/ballerina-standard-library/issues/841#issuecomment-998575456

Tested the fix locally and it passes successfully.
## Examples
n/a
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/RequestCompleted.java [hunk 1]
```diff
@@ -83,7 +83,9 @@ public void readInboundResponseHeaders(ChannelHandlerContext ctx, Http2HeadersFr
     public void readInboundResponseBody(ChannelHandlerContext ctx, Http2DataFrame http2DataFrame,
                                         OutboundMsgHolder outboundMsgHolder, boolean serverPush,
                                         Http2MessageStateContext http2MessageStateContext) {
-        LOG.warn("readInboundResponseBody is not a dependant action of this state");
+        http2MessageStateContext.setSenderState(new ReceivingEntityBody(http2TargetHandler, http2RequestWriter));
```
Comment (chamil321) [inline native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/RequestCompleted.java:86]: Shall we add a comment saying, following is needed for the bidirectional streaming?

**Review by chamil321 — APPROVED** [review]

File: native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/RequestCompleted.java (hunk 1 above)
Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/RequestCompleted.java:88]: Added in https://github.com/ballerina-platform/module-ballerina-http/pull/796/commits/ce5400904b192a4bff75ecbe68f5d1402e8539da

**Review by chamil321 — APPROVED** [review]

**Comment by dilanSachi** [comment]
This fix will be analyzed further and decide on a solution.

**Comment by dilanSachi** [comment]
This is tracked in https://github.com/ballerina-platform/ballerina-standard-library/issues/841

**Pull request merged**